src/linear_samplers/block_col_rand_cyclic.jl

```diff
@@ -0,0 +1,92 @@
+# This code was written by Nathaniel Pritchard
+"""
+    LinSysBlkColRandCyclic <: LinSysBlkColSampler 
+
+A mutable structure with fields to handle randomly permuted block sampling. Can allow for fixed
+blocks or blocks whose entries are randomly permuted. After each cycle, a new random ordering is
```

Comment:
What does it mean for blocks whose entries are randomly permuted? So within a block, you permute the entries? This needs to be clarified.

Reply:
@vp314 this is a good point. In my mind, if this code is something that can be just copied and pasted and put into a function, and then called from the two sampler routines, we should do it now. The rationale is that it will keep the code clean, and minimize maintenance - its always a good practice to modularize code that is repeated in various places. If it requires something more sophisticated maybe we can leave it for later.

Reply:
I have removed this wording.